sys/include/xtimer.h

```diff
@@ -184,120 +189,127 @@ static inline void xtimer_tsleep64(xtimer_ticks64_t ticks);
+ * The message struct specified by msg parameter will not be copied, e.g., it
  * needs to point to valid memory until the message has been delivered.
  *
- * @param[in] timer         timer struct to work with.
- *                          Its xtimer_t::target and xtimer_t::long_target
- *                          fields need to be initialized with 0 on first use.
- * @param[in] offset        microseconds from now
- * @param[in] msg           ptr to msg that will be sent
- * @param[in] target_pid    pid the message will be sent to
+ * @param[in] timer        timer struct to work with.
+ *                         Its xtimer_t::target and xtimer_t::long_target
+ *                         fields need to be initialized with 0 on first use.
+ * @param[in] offset_us    microseconds from now
+ * @param[in] msg          ptr to msg that will be sent
+ * @param[in] target_pid   pid the message will be sent to
  */
-static inline void xtimer_set_msg(xtimer_t *timer, uint32_t offset, msg_t *msg, kernel_pid_t target_pid);
+static inline void xtimer_set_msg(xtimer_t *timer, uint32_t offset_us,
+                                  msg_t *msg, kernel_pid_t target_pid);
 
 /**
  * @brief Set a timer that sends a message, 64bit version
  *
- * This function sets a timer that will send a message @p offset microseconds
- * from now.
+ * This function sets a timer that will send a message @p offset_us
+ * microseconds from now.
  *
- * The mesage struct specified by msg parameter will not be copied, e.g., it
+ * The message struct specified by msg parameter will not be copied, e.g., it
  * needs to point to valid memory until the message has been delivered.
  *
- * @param[in] timer         timer struct to work with.
- *                          Its xtimer_t::target and xtimer_t::long_target
- *                          fields need to be initialized with 0 on first use.
- * @param[in] offset        microseconds from now
- * @param[in] msg           ptr to msg that will be sent
- * @param[in] target_pid    pid the message will be sent to
+ * @param[in] timer        timer struct to work with.
+ *                         Its xtimer_t::target and xtimer_t::long_target
+ *                         fields need to be initialized with 0 on first use.
+ * @param[in] offset_us    microseconds from now
+ * @param[in] msg          ptr to msg that will be sent
+ * @param[in] target_pid   pid the message will be sent to
  */
-static inline void xtimer_set_msg64(xtimer_t *timer, uint64_t offset, msg_t *msg, kernel_pid_t target_pid);
+static inline void xtimer_set_msg64(xtimer_t *timer, uint64_t offset_us,
+                                    msg_t *msg, kernel_pid_t target_pid);
 
 /**
  * @brief Set a timer that wakes up a thread
  *
- * This function sets a timer that will wake up a thread when the timer has
- * expired.
+ * This function sets a timer to wakeup a thread after @p offset_us microseconds.
+ * The calling or set thread is not put to sleep.
```

Comment:
That is also not, what was stated before. I guess a better wording is "The calling or set thread is not *guaranteed* to be put to sleep", but a) I'm no expert here, b) I'm not sure why the old wording is insufficient to you.

Reply:
The calling thread is guaranteed to not be put to sleep since there is no call to thread_sleep in the implementation. But this is about as relevant as stating that the calling thread is guaranteed to not launch a web server, it has nothing to do with the actual function being described.
If the original wording is unclear, how about describing some more of the internals: "set a timer that will call thread_wakeup on the given pid when it expires"
But I don't really see any difference to what the original text says.

Reply:
@gebart obviously two people did not fully grasp what the function does, by only reading this description. Just rewording the statement does not help as it does not contain any additional information. So maybe it needs some clarification in what the function does not, as a user might assume it would. @miri64 That’s why i added this information.  
Additionally, you could argue that this description is not needed at all as it only contains the same information as the brief description. 
This said, I will set it back to the original statement as it seems to satisfy the amount of information a user needs.

Reply:
> @gebart obviously two people did not fully grasp what the function does, by only reading this description.

For every description in existence you'll find two people that do not fully grasp what the corresponding function does.

Reply:
Sorry, I was a bit harsh and unfriendly in my previous comment, I shouldn't have posted that. I see you point that there is value in changing the description since the user don't understand it as it is written now.

Reply:
@gebart  all good.

I tried to put what we discussed in the discription.  Could you reassess it?